### ISSUE TITLE
Flaky test: CentralDogmaBeanTest.test()

### ISSUE THREAD

**Issue by trustin** [line/centraldogma#280]
```
com.linecorp.centraldogma.it.updater.CentralDogmaBeanTest > test FAILED
    org.junit.ComparisonFailure: expected:<[10]> but was:<[21]>
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at com.linecorp.centraldogma.it.updater.CentralDogmaBeanTest.test(CentralDogmaBeanTest.java:77)
```

May be related with #276 /cc @jmostella

**Comment by jmostella** [comment]
I tried the test locally, and the build consistently passed with the latest `master` Under what conditions do you see a failure?

**Comment by jmostella** [comment]
After some analyzing the code, I believe I’ve found the issue is a race condition. 

Since the test `updateListenerIgnoreDefault()` was added it is now being run before `test()`.
This is leaving the  `@CentralDogmaBean(project = "a", repository = "b", path = "/c.json")` bean in "dirty" state on the server side for `test()` run.
This is not an issue for the block of code at:

https://github.com/line/centraldogma/blob/master/it/src/test/java/com/linecorp/centraldogma/it/updater/CentralDogmaBeanTest.java#L77-L80

**Given** the `FileWatcher` doesn’t return quickly. I don't want to speculate as to why we see this more consistently on the [AppVeyor CI build](https://ci.appveyor.com/project/line/centraldogma/builds/19846278) (may have something to do with environment?). 

Since this behavior is allowable by design, I will simply remove the verification for clean starting state in the fix PR #286

**Comment by trustin** [comment]
Much appreciated, @jmostella :100:
